scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1046,66 +1046,60 @@
+
 (timers/deftimer [cook-mesos scheduler sort-gpu-jobs-hierarchy-duration])
 
-(defn sort-gpu-jobs-by-dru
+(defn- sort-gpu-jobs-by-dru
   "Return a list of gpu job entities ordered by dru"
   [pending-task-ents running-task-ents user->dru-divisors]
-  (let [jobs (timers/time!
-               sort-gpu-jobs-hierarchy-duration
-               (->> (into (vec running-task-ents) pending-task-ents)
-                    (group-by util/task-ent->user)
-                    (pc/map-vals (fn [task-ents] (into (sorted-set-by (util/same-user-task-comparator)) task-ents)))
-                    (dru/sorted-task-cumulative-gpu-score-pairs user->dru-divisors)
-                    (filter (fn [[task _]] (contains? pending-task-ents task)))
-                    (map (fn [[task _]] (:job/_instance task)))))]
-    jobs))
+  (sort-jobs-by-dru-helper pending-task-ents running-task-ents user->dru-divisors
+                           dru/sorted-task-cumulative-gpu-score-pairs sort-gpu-jobs-hierarchy-duration))
 
 (defn sort-jobs-by-dru
   "Returns a map from job category to a list of job entities, ordered by dru"
-  [filtered-db unfiltered-db]
+  [unfiltered-db]
   ;; This function does not use the filtered db when it is not necessary in order to get better performance
   ;; The filtered db is not necessary when an entity could only arrive at a given state if it was already committed
   ;; e.g. running jobs or when it is always considered committed e.g. shares
   ;; The unfiltered db can also be used on pending job entities once the filtered db is used to limit
   ;; to only those jobs that have been committed.
-  (let [pending-job-ents-by (group-by util/categorize-job (util/get-pending-job-ents unfiltered-db))
-        pending-task-ents-by (reduce-kv (fn [m category pending-job-ents]
-                                          (assoc m category
-                                                   (into #{}
-                                                         (map util/create-task-ent)
-                                                         pending-job-ents)))
-                                        {}
-                                        pending-job-ents-by)
-        running-task-ents-by (group-by (comp util/categorize-job :job/_instance)
-                                       (util/get-running-task-ents unfiltered-db))
-        user->dru-divisors (share/create-user->share-fn unfiltered-db)]
-    {:normal (sort-normal-jobs-by-dru (:normal pending-task-ents-by)
-                                      (:normal running-task-ents-by)
-                                      user->dru-divisors)
-     :gpu (sort-gpu-jobs-by-dru (:gpu pending-task-ents-by)
-                                (:gpu running-task-ents-by)
-                                user->dru-divisors)}))
+  (let [category->pending-job-ents (group-by util/categorize-job (util/get-pending-job-ents unfiltered-db))
+        category->pending-task-ents (reduce-kv (fn [m category pending-job-ents]
```

Comment:
I think this can just be:

```clojure
category->pending-task-ents (map-vals #(map util/create-task-ent pending-job-ents) category->pending-job-ents)
```